agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -0,0 +1,601 @@
+	parsedARN, err := arn.Parse(credentialspecSSMARN)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+
+	ssmClient := cs.ssmClientCreator.NewSSMClient(cs.region, iamCredentials)
+
+	ssmParam := filepath.Base(parsedARN.Resource)
+	ssmParams := []string{ssmParam}
+
+	ssmParamMap, err := ssm.GetParametersFromSSM(ssmParams, ssmClient)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+
+	ssmParamData := ssmParamMap[ssmParam]
+
+	localCredSpecFilePath := fmt.Sprintf("%s/ssm_%s_%s.json", cs.credentialSpecResourceLocation, cs.taskARN, ssmParam)
+
+	err = cs.writeSSMFile(ssmParamData, localCredSpecFilePath)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+
+	dockerHostconfigSecOptCredSpec := fmt.Sprintf("credentialspec=file://%s", filepath.Base(localCredSpecFilePath))
+	cs.updateCredSpecMapping(originalCredentialspec, dockerHostconfigSecOptCredSpec)
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) writeS3File(writeFunc func(file oswrapper.File) error, filePath string) error {
+	temp, err := cs.ioutil.TempFile(cs.credentialSpecResourceLocation, tempFileName)
+	if err != nil {
+		return err
+	}
+	defer temp.Close()
+
+	err = writeFunc(temp)
+	if err != nil {
+		return err
+	}
+
+	err = temp.Chmod(os.FileMode(filePerm))
+	if err != nil {
+		return err
+	}
+
+	// Persist the file to disk.
+	err = temp.Sync()
+	if err != nil {
+		return err
+	}
+
+	err = cs.os.Rename(temp.Name(), filePath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) writeSSMFile(ssmParamData, filePath string) error {
+	return cs.ioutil.WriteFile(filePath, []byte(ssmParamData), filePerm)
+}
+
+func (cs *CredentialSpecResource) getCredSpecMap() map[string]string {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	return cs.CredSpecMap
+}
+
+func (cs *CredentialSpecResource) GetTargetMapping(credSpecInput string) (string, error) {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+
+	targetCredSpecMapping, ok := cs.CredSpecMap[credSpecInput]
+	if !ok {
+		return "", errors.New("unable to obtain credentialspec mapping")
+	}
+
+	return targetCredSpecMapping, nil
+}
+
+func (cs *CredentialSpecResource) updateCredSpecMapping(credSpecInput, targetCredSpec string) {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	cs.CredSpecMap[credSpecInput] = targetCredSpec
+}
+
+// Cleanup removes the credentialspec created for the task
+func (cs *CredentialSpecResource) Cleanup() error {
+	cs.clearCredentialSpec()
+	return nil
+}
+
+// clearCredentialSpec cycles through the collection of credentialspec data and
+// removes them from the task
+func (cs *CredentialSpecResource) clearCredentialSpec() {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+
+	for key, value := range cs.CredSpecMap {
+		credSpecSplit := strings.SplitAfterN(value, "credentialspec=", 2)
+		localCredentialSpecFile := credSpecSplit[1]
```

Comment:
Shouldn't delete the Credentialspec with file option

Reply:
Done.